HomeTask_10_dir/HomeTask_10.py

```diff
@@ -0,0 +1,10 @@
+import re
+with open('input.txt', 'r') as inp:
+    data = inp.read()
+    regexp = re.findall(r'(?<![A-Z]{4})(?<=[A-Z]{3})[a-z](?<![A-Z])(?=[A-Z]{3})(?![A-Z]{4})', data)
```

Comment:
Some things are a little bit  redundant here you able to find more compact solution just using (?<= ..) and (?=...)